models/userDao.js

```diff
@@ -0,0 +1,77 @@
+const { appDataSource } = require("./appDataSource");
+//유저 생성
+const createUser = async (
+  email,
+  password,
+  name,
+  birth,
+  phoneNumber,
+  address
+) => {
+  try {
+    return await appDataSource.query(
+      `INSERT INTO users(
+          email,
+          password,
+          name,
+          birth,
+          phone_number,
+          address)
+      VALUES(
+          ?,?,?,?,?,?
+      )`,
+      [email, password, name, birth, phoneNumber, address]
+    );
+  } catch (err) {
+    err.message;
+    err.statusCode = 400;
+    throw err;
+  }
+};
+
+//유저 이메일 중복 확인
+const getIdByUserEmail = async (email) => {
```

Comment:
지금같이 단순 존재 여부만을 파악할 때에는 db query를 통해 실제 어떠한 데이터를 받아오는 것 보다 SELECT EXIST 를 활용하여 참, 거짓만 판단하는게 보다 더 효율적입니다. (MySQL에서 참과 거짓은 boolean이 아닌 1, 0으로 리턴됩니다.)

```javascript
const result = async (email) => {
  const [result] = appDataSource.query(
    `SELECT EXISTS (
        SELECT
          id
        FROM users
        WHERE email = ?
      ) as registerd`,
    [email]
  );
  return !!parseInt(result.registerd);        // 0, 1로 나타내어지는 result.registerd를 double bang (!!)을 붙여 booltype으로 변환하여 return합니다.
};
```

위의 피드백에 따라 dao이름도 아래와 같이 바뀌면 적절할 것 같습니다!

```suggestion
const checkRegisterdEmail = async (email) => {
```